pages/authorize.js

```diff
@@ -0,0 +1,198 @@
+// Module imports
+import { bindActionCreators } from 'redux'
+import React from 'react'
+
+
+
+
+
+// Component imports
+import { actions } from '../store'
+import Component from '../components/Component'
+import Page from '../components/Page'
+
+
+
+
+
+// Component constants
+const title = 'Authorize Application'
+
+
+
+
+
+class Authorize extends Component {
+
+  /***************************************************************************\
+   Public Methods
+   \***************************************************************************/
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      clientName: null,
+      scopes: [],
+      scope: '',
+      transactionId: '',
+      submitting: false
+    }
+  }
+
+  static async getInitialProps ({ query }) {
+    if (query.redirectUri) {
+      this.setState({
+        redirectUri: query.redirectUri
+      })
+    }
+    return query
+  }
+
+  async componentDidMount () {
+    let { client_id, state, scope, response_type } = this.props
+
+    if (client_id && state && scope && response_type) {
+      try {
+        let token = localStorage.getItem('access_token')
+        let response = await fetch(`/api/oauth2/authorize?client_id=${client_id}&scope=${scope}&state=${state}&response_type=${response_type}`, {
```

Comment:
I'd like to refactor this into an action at some point, but I'll let it be for now so we can get this shit done.